### DISCUSSION TITLE
Mime type error

### DISCUSSION THREAD

**Discussion by CodyPin** [pocketbase/pocketbase#1086] [Q&A] [answered]
First time posting, not sure if any of this formatting is correct at all, or maybe I should make a issue in the dark sdk repo? I am not sure, apologize in advance.

I have been building this flutter project for a few months now and had been using pocketbase for my backend. Everything is going well so far except when I try to upload an image.

`if (image?.path != null) {
        print(lookupMimeType(image?.path ?? ''));
        await client.collection('coupons').create(
          body: body,
          files: [
            http.MultipartFile.fromString(
              'image',
              'png',
              filename: image?.path,
            ),
          ],
        );
      } else {
        await client.collection('coupons').create(
              body: body,
            );
      }`

Above is the code snipit where I call the api to upload a record to a 'coupons' table, without the image it works via the else part of the code. the print on line 2 and the error is

`I/flutter (11050): image/png
I/flutter (11050): ClientException: {url: http://192.168.0.100/api/collections/coupons/records, isAbort: false, statusCode: 400, response: {code: 400, message: Failed to create record., data: {image: {code: validation_invalid_mime_type, message: The following mime types are only allowed: image/jpg,image/jpeg,image/png,image/svg+xml,image/gif.}}}, originalError: null}`

So what I understand from this is that the system knows it is a png, but for some reason pocketbase refuse to reconigse this file as a png?

I also tried removing the mime type restraint, the upload is 'successful' with 
![brave_Pcf3jUjen5](https://user-images.githubusercontent.com/80540276/203381596-e3d34018-ef86-4a2a-89c6-01b205976b30.png)
as the image.

I was using an older version of pocketbase and I just update it to 0.8.0 today but still no luck. I know that pocketbase is still relatively new so I cannot find anyone with a simliar problem so far. Does anyone know what I did wrong?

**Comment by CodyPin** [comment]
Sorry about the mess up in the coding bit, here is a touch up version
```
if (image?.path != null) {
     print(lookupMimeType(image?.path ?? ''));
     await client.collection('coupons').create(
       body: body,
       files: [
         http.MultipartFile.fromString(
           'image',
           'png',
           filename: image?.path,
         ),
       ],
     );
   } else {
     await client.collection('coupons').create(
           body: body,
         );
   }
```

```
I/flutter (11050): image/png
I/flutter (11050): ClientException: {url: http://192.168.0.100/api/collections/coupons/records, isAbort: false, statusCode: 400, response: {code: 400, message: Failed to create record., data: {image: {code: validation_invalid_mime_type, message: The following mime types are only allowed: image/jpg,image/jpeg,image/png,image/svg+xml,image/gif.}}}, originalError: null}
```

**Comment by ganigeorgiev** [comment, marked as answer]
The `http.MultipartFile.fromString` creates a new multipart file with **content** the provided string, aka. in your case that would be just the "png" text. More info you could find in the official dart docs - https://pub.dev/documentation/http/latest/http/MultipartFile/MultipartFile.fromString.html.

---

If you want to create a multipart file from a file **path**, there is a static method [`fromPath`](https://pub.dev/documentation/http/latest/http/MultipartFile/fromPath.html):
```dart
http.MultipartFile.fromPath("yourFileFieldName", image?.path, filename: "my-image-filename")
```

_(the above works only in `dart:io` supported environments)_

---

Note that `http.MultipartFile` has also other constructors available (eg. `fromBytes`). For more information you could refer to the official dart http docs - https://pub.dev/documentation/http/latest/http/MultipartFile-class.html.

**Reply by CodyPin** [reply]
Thx a lot, it works! In hindsight I should have just looked into the http docs, you are a life saver!
